### PR TITLE
Round 12

### LINKED ISSUE #219: Test: Resolve calcs and merge them in margin
Not sure of the category to put this under

```md
# Resolve calcs and merge them in margin

Resolve any static calculations, then apply merging of identical values for
properties like `margin`, `padding`, or `border`.
```

```css
a {
  margin: calc(1px + 2px) calc(4px - 1px);
}
```

```css
a{margin:3px}
```

[CSS Minifier Playground](https://thejaredwilcurt.com/playground/?v=eNpLVKjmUlDITSxKz8yzUkhOzEnWMCyoUNBWMCqo0ITwTYB8XQWgqCZXLQBPgw3O&t=eNpTVghKLc7PKUtVSE7MSS5WSMxLAQA%2FAwad&d=eNoDAAAAAAE%3D&x=eNpLrM5NLErPzLMyLqioBQAkjwUt&c=18)

Minifier  | Version  | 🏆  | Size | Output
:--       | :--:     | :-- | :--  | :--
lightning | v1.33.0  | 🏆  | 13   | `a{margin:3px}`
cssnano   | v8.0.8   | 🏆  | 13   | `a{margin:3px}`
esbuild   | v0.28.2  | 🏆  | 13   | `a{margin:3px}`
csslop    | v0.0.26  | ❌️  | 16   | `a{margin:3px3px}`
sass      | v1.103.1 |     | 17   | `a{margin:3px 3px}`
csskit    | v0.0.31  |     | 40   | `a{margin:calc(1px + 2px)calc(4px - 1px)}`
cleancss  | v5.3.3   |     | 41   | `a{margin:calc(1px + 2px) calc(4px - 1px)}`
csso      | v5.0.5   |     | 41   | `a{margin:calc(1px + 2px) calc(4px - 1px)}`

### LINKED ISSUE #246: Test: Remove `-webkit-box-flex` and `-ms-flex` when `flex` is present
We should probably just have a `vendors/xxxx` category for stuff like this to be moved to.

`values/00xx`

```md
# Remove `-webkit-box-flex` and `-ms-flex` when `flex` is present

The vendor prefix `-webkit-box-flex` and `-ms-flex` are no longer needed, all
browsers support `flex` if it is provided.
```

```css
a {
  -webkit-box-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-left: 1rem;
  padding-right: 1rem;
}
```

```css
a{flex:1;padding-left:1rem;padding-right:1rem}
```

[CSS Minifier Playground](https://thejaredwilcurt.com/playground/?v=eNpLVKjmUlDQLU9Nys4s0U3Kr9BNy0mtsFIwtAYJ5xYjc5GYBYkpKZl56bo5qWklQKGi1Fxk0aLM9Ay4cC0ArXkdyg%3D%3D&x=eNpLrE7LSa2wMrQuSExJycxL181JTSuxMixKzYWLFGWmZ0CEagGPmBEZ&t=eNpTVghKzc0vS1VI0C1PTcrOLNFNyq%2FQTctJrUhQKM9IzVNIgLALilKLU%2FNKAImhERk%3D&d=eNoty8ENhCAQBdD7VvELWKzGApDw0YmEIQMq5auJ9%2FfmjThZohqqMcmAdxfDLt0FHS5lDg9pKIqsZaWhkJHxjyVnBNOr0RraUata%2F%2FkvJEh%2FWzU95eHTDcowJe0%3D&c=32)

Minifier  | Version  | 🏆  | Size | Output
:--       | :--:     | :-- | :--  | :--
csslop    | v0.0.36  |     | 65   | `a{-webkit-box-flex:1;flex:1;padding-left:1rem;padding-right:1rem}`
lightning | v1.33.0  | 🏆  | 46   | `a{flex:1;padding-left:1rem;padding-right:1rem}`
cleancss  | v5.3.3   |     | 76   | `a{-webkit-box-flex:1;-ms-flex:1;flex:1;padding-left:1rem;padding-right:1rem}`
cssnano   | v9.0.3   | 🏆  | 46   | `a{flex:1;padding-left:1rem;padding-right:1rem}`
csskit    | v0.0.32  |     | 76   | `a{-webkit-box-flex:1;-ms-flex:1;flex:1;padding-left:1rem;padding-right:1rem}`
csso      | v5.0.5   |     | 76   | `a{-webkit-box-flex:1;-ms-flex:1;flex:1;padding-left:1rem;padding-right:1rem}`
esbuild   | v0.28.2  |     | 76   | `a{-webkit-box-flex:1;-ms-flex:1;flex:1;padding-left:1rem;padding-right:1rem}`
sass      | v1.104.0 |     | 76   | `a{-webkit-box-flex:1;-ms-flex:1;flex:1;padding-left:1rem;padding-right:1rem}`

### PR DESCRIPTION
**Pull request by TheJaredWilcurt**
* [x] Cleaned up the `duplicates` tests, adding `alt`s as needed for things not related to the idea being tested.
* [x] Created a "vendors" category
  * [x] Moved `duplicates/0008` to `vendors/0001`
  * [x] Closes #246
* [x] Created a "calc" category
  * [x] Closes #219
  * [x] Moved `values/0011` to `calc/0002`
  * [x] Moved `values/0014` to `calc/0003`
  * [x] Moved `values/0015` to `calc/0004`
  * [x] Moved `values/0016` to `calc/0005`
  * [x] Moved `values/0017` to `calc/0006`
  * [x] Moved `values/0021` to `calc/0007`
  * [x] Moved `values/0039` to `calc/0008`
  * [x] Moved `values/0040` to `calc/0009`
  * [x] Moved `values/0041` to `calc/0010`
  * [x] Moved `values/0042` to `calc/0011`
  * [x] Moved `values/0043` to `calc/0012`
  * [x] Moved `values/0044` to `calc/0013`
  * [x] Moved `values/0045` to `calc/0014`
  * [x] Moved `values/0046` to `calc/0015`
  * [x] Moved `values/0047` to `calc/0016`
  * [x] Moved `values/0048` to `calc/0017`
  * [x] Moved `values/0049` to `calc/0018`
  * [x] Moved `values/0050` to `calc/0019`
  * [x] Moved `values/0051` to `calc/0020`
  * [x] Moved `values/0053` to `calc/0021`
  * [x] Moved `values/0054` to `calc/0022`
  * [x] Moved `values/0055` to `calc/0023`
  * [x] Moved `values/0056` to `calc/0024`
  * [x] Moved `values/0069` to `calc/0025`
* [x] Since a lot of existing tests were moved to different categories, I moved the highest test numbers up to fill in the gaps.
  * [x] Moved `duplicates/0024` to `duplicates/0008`
  * [x] Moved `values/0078` to `values/0011`
  * [x] Moved `values/0077` to `values/0014`
  * [x] Moved `values/0076` to `values/0015`
  * [x] Moved `values/0075` to `values/0016`
  * [x] Moved `values/0074` to `values/0017`
  * [x] Moved `values/0073` to `values/0021`
  * [x] Moved `values/0072` to `values/0039`
  * [x] Moved `values/0071` to `values/0040`
  * [x] Moved `values/0068` to `values/0041`
  * [x] Moved `values/0067` to `values/0042`
  * [x] Moved `values/0066` to `values/0043`
  * [x] Moved `values/0065` to `values/0044`
  * [x] Moved `values/0064` to `values/0045`
  * [x] Moved `values/0063` to `values/0046`
  * [x] Moved `values/0062` to `values/0047`
  * [x] Moved `values/0061` to `values/0048`
  * [x] Moved `values/0060` to `values/0049`
  * [x] Moved `values/0059` to `values/0051`
  * [x] Moved `values/0058` to `values/0053`
  * [x] Moved `values/0057` to `values/0054`
* [x] `values/0070` was in wrong category, moved to `property/0008`
* [x] `values/0049` (formerly `values/0060`) had a correct test but the description was technically wrong. Updated the description and added a new test `values/0050` to handle the edge case (that CSSLOP was failing).
* [x] Update dependencies
* [x] Update results

### CODE REVIEW

File: tests/duplicates/0001/source.css [hunk 1]
```diff
@@ -1,4 +1,4 @@
 a {
   color: red;
-  color: blue;
```
Comment (TheJaredWilcurt) [inline tests/duplicates/0001/source.css:3]: These tests are about de-duping, not about converting to hex. changing `blue` to `tan` to keep them more focused.

File: tests/duplicates/0003/expected.css [hunk 2]
```diff
@@ -1 +1 @@
-b{margin:0}a{color:#00f;font-size:16px}
```
Comment (TheJaredWilcurt) [inline tests/duplicates/0003/expected.css:1]: This test has a lot of overlap with the tests in `merging`. Added alts to match those tests. We care about de-duping, not enforcing rule/property orders in this test.

File: tests/duplicates/0006/validate.html [hunk 3]
```diff
@@ -1,26 +1,34 @@
...
+          .expected a{color:red!important}
+        </style>
+        <div class="expected">
+          <a>Test element</a>
+        </div>
+      </div>
+    </div>
+    <p>Both elements should be red. The !important declaration wins over the later non-important one.</p>
+  </body>
 </html>
```
Comment (TheJaredWilcurt) [inline tests/duplicates/0006/validate.html:34]: Only indentation changes.

File: tests/duplicates/0007/expected.alt.css [hunk 4]
```diff
@@ -0,0 +1 @@
+a{font-size:16px;color:red}
```
Comment (TheJaredWilcurt) [inline tests/duplicates/0007/expected.alt.css:1]: Property order no longer enforced in this test. Accepts:

* `a{color:red;font-size:16px}`
* `a{font-size:16px;color:red}`

File: tests/duplicates/0009/expected.alt.css [hunk 5]
```diff
@@ -0,0 +1 @@
+.footer,.nav .item{color:red}
```
Comment (TheJaredWilcurt) [inline tests/duplicates/0009/expected.alt.css:1]: Selector order not pertinent to de-duping. Now accepts:

* `.nav .item,.footer{color:red}`
* `.footer,.nav .item{color:red}`

File: tests/duplicates/0010/expected.alt.css [hunk 6]
```diff
@@ -0,0 +1 @@
+h2,h1{color:red}
```
Comment (TheJaredWilcurt) [inline tests/duplicates/0010/expected.alt.css:1]: Selector order not pertinent to de-duping. Now accepts:

* `h1,h2{color:red}`
* `h2,h1{color:red}`

File: tests/duplicates/0012/expected.alt.css [hunk 7]
```diff
@@ -0,0 +1 @@
+a{font-size:1rem;color:red}
```
Comment (TheJaredWilcurt) [inline tests/duplicates/0012/expected.alt.css:1]: Property order no longer enforced:

* `a{color:red;font-size:1rem}`
* `a{font-size:1rem;color:red}`

File: tests/duplicates/0023/expected.alt.css [hunk 8]
```diff
@@ -0,0 +1 @@
+body{background:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAkAAAAGCAIAAACepSOSAAAAIklEQVR42mP4DwZpxjMxGQwQCojQGAg5AvowteK1D4hxIQApumnK5mwMyQAAAABJRU5ErkJggg==);image-rendering:pixelated;mask-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAkAAAAGCAIAAACepSOSAAAAIklEQVR42mP4DwZpxjMxGQwQCojQGAg5AvowteK1D4hxIQApumnK5mwMyQAAAABJRU5ErkJggg==)}
```
Comment (TheJaredWilcurt) [inline tests/duplicates/0023/expected.alt.css:1]: Property order updated to support alphabetized properties, which for some reason CSSNano has enabled in its advanced preset.

File: tests/values/0049/README.md [hunk 9]
```diff
@@ -1,4 +1,8 @@
...
+# url() quotes required when URL contains 2 or more parentheses
 
-`calc(100px * 0)` multiplies a dimension by zero. The result is `0px`, which
-serializes as `0`.
+Removing quotes from `url("image (1).png")` would produce `url(image (1).png)`
+which is a parse error -- the unquoted form cannot contain `(` or `)`. You could
+escape these characters to avoid a syntax error: `url(image\ \(1\).png)`,
+however, that removes two characters (quotes) and adds in 3 (slashes), in this
+case. If there is only one opening or one closing parenthesis, then escaping
+would be shorter.
```
Comment (TheJaredWilcurt) [inline tests/values/0049/README.md:8]: This title & description was elongated to better explain edge cases, and `values/0050` is a new test to handle the edge case that CSSLOP failed. the `0049` source/expected did not change.

File: tests/vendors/0001/source.css [hunk 10]
```diff
@@ -0,0 +1,7 @@
+a {
+  -webkit-transform: rotate(45deg);
+  -moz-transform: rotate(45deg);
+  -ms-transform: rotate(45deg);
+  -o-transform: rotate(45deg);
+  transform: rotate(45deg);
+}
```
Comment (TheJaredWilcurt) [inline tests/vendors/0001/source.css:7]: Added additional vendor prefixes to remove, CSSLOP now fails this test.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -201,13 +204,13 @@ In `lib/loaders/loadAllMinifiers.js`:
...
 
 export const registry = Object.freeze({
-  'my-minifier': {
-    title: 'My Minifier',
-    url: 'https://my-minifier.github.io/playground',
-    minify: myMinifier
+  'some-minifier': {
+    title: 'Some Minifier',
+    url: 'https://some-minifier.github.io/playground',
+    minify: someMinifier
```
Comment (TheJaredWilcurt) [inline CONTRIBUTING.md:213]: Renamed `my` to `some` to match above examples.

**Review by keithamus — APPROVED** [review]
Nice work!
